src/Provider/AbstractProvider.php

```diff
@@ -289,68 +320,63 @@ public function getAccessToken($grant = 'authorization_code', array $params = []
             'grant_type'    => (string) $grant,
         ];
 
-        $requestParams = $grant->prepRequestParams($defaultParams, $params);
+        $requestParams  = $this->httpBuildQuery($grant->prepRequestParams($defaultParams, $params));
+        $urlAccessToken = $this->urlAccessToken();
 
-        try {
-            $client = $this->getHttpClient();
-            switch (strtoupper(static::ACCESS_TOKEN_METHOD)) {
-                case 'GET':
-                    // @codeCoverageIgnoreStart
-                    // No providers included with this library use get but 3rd parties may
-                    $httpResponse = $client->get(
-                        $this->urlAccessToken(),
-                        $this->getHeaders(),
-                        $requestParams
-                    );
-                    $response = (string) $httpResponse->getBody();
-                    break;
-                    // @codeCoverageIgnoreEnd
-                case 'POST':
-                    $httpResponse = $client->post(
-                        $this->urlAccessToken(),
-                        $this->getHeaders(),
-                        $requestParams
-                    );
-                    $response = (string) $httpResponse->getBody();
-                    break;
+        switch (strtoupper(static::ACCESS_TOKEN_METHOD)) {
+            case 'GET':
                 // @codeCoverageIgnoreStart
-                default:
-                    throw new \InvalidArgumentException('Neither GET nor POST is specified for request');
+                // No providers included with this library use get but 3rd parties may
+                $urlAccessToken .= (strpos($urlAccessToken, '?') ? '&' : '?') . $requestParams;
```

Comment:
> parsing a query string

You're parsing the URL, which of course wouldn't be more efficient than a string search (micro-optimization anyway). It's just more reliable I would think, than to assume that the presence of a '?' guarantees we should use &. What if I use 'http://test.com/?'? We don't want 'http://test.com/?&param=value.